core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/stack/termination/CleanupFreeIpaTask.java

```diff
@@ -59,5 +63,14 @@ public void run() {
         } catch (Exception e) {
             LOGGER.error("FreeIPA cleanup failed", e);
         }
+        // Cleanup vault entries
+        fqdns.forEach((String fqdn) -> {
```

Comment:
this logic should be moved to freeipa CleanupService, as the request has all the necessary info, and invoking FreeIPA mgmt service for every host is not ideal

Reply:
I moved it to the `CleanupService` which allows some of the changes to be simplified.
But I am not exactly sure how to know which hosts are the "CM" hosts. I tried to get the gateway like in `KeytabConfigurationHandler` and `KeytabProvider` (these generate the CM keytab/secret), but when this is run on the freeipa `CleanupService`, it tries to delete the gateway which ends up being `ipaserver0.cloudera.site` and that doesn't work since it is the IPA server. I am not sure if there is a better way to filter the contents or not. If not, maybe we just run the delete for nodes which are no-ops.

Reply:
calling delete for all hosts is not a problem